integration-tests/emulated/src/tests/spiritnet/sudo.rs

```diff
@@ -0,0 +1,163 @@
+use parity_scale_codec::Encode;
+use runtime_common::AccountId;
+use xcm::{v3::WeightLimit, DoubleEncoded, VersionedMultiLocation, VersionedXcm};
+use xcm_emulator::{
+	assert_expected_events, Here,
+	Instruction::{BuyExecution, Transact, UnpaidExecution, WithdrawAsset},
+	Junction, Junctions, OriginKind, Parachain, ParentThen, RelayChain, TestExt, Weight, Xcm,
+};
+
+#[test]
+fn test_sudo_call_from_relay_chain_to_spiritnet() {
+	MockNetworkPolkadot::reset();
+
+	let code = vec![];
+
+	let call: DoubleEncoded<()> = <Spiritnet as Parachain>::RuntimeCall::System(frame_system::Call::set_code { code })
+		.encode()
+		.into();
+	let sudo_origin = <Polkadot as RelayChain>::RuntimeOrigin::root();
+	let parachain_destination = Polkadot::child_location_of(spiritnet::PARA_ID.into()).into_versioned();
+
+	let weight_limit = WeightLimit::Unlimited;
+	let require_weight_at_most = Weight::from_parts(1600000000000, 200000);
+	let check_origin = None;
+	let origin_kind = OriginKind::Superuser;
+
+	// the relay chain would submit an unpaid execution request.
+	let xcm = VersionedXcm::from(Xcm(vec![
+		UnpaidExecution {
+			weight_limit,
+			check_origin,
+		},
+		Transact {
+			origin_kind,
+			require_weight_at_most,
+			call,
+		},
+	]));
+
+	//Send XCM message from relay chain
+	Polkadot::execute_with(|| {
+		assert_ok!(<Polkadot as PolkadotPallet>::XcmPallet::send(
+			sudo_origin,
+			Box::new(parachain_destination),
+			Box::new(xcm)
+		));
+
+		type RuntimeEvent = <Polkadot as RelayChain>::RuntimeEvent;
+
+		assert_expected_events!(
+			Polkadot,
+			vec![
+				RuntimeEvent::XcmPallet(pallet_xcm::Event::Sent { .. }) => {},
+			]
+		);
+	});
+
+	Spiritnet::execute_with(|| {
+		type SpiritnetRuntimeEvent = <Spiritnet as Parachain>::RuntimeEvent;
+		assert_expected_events!(
+			Spiritnet,
+			vec![
+				SpiritnetRuntimeEvent::DmpQueue(cumulus_pallet_dmp_queue::Event::ExecutedDownward {
+					outcome: xcm::v3::Outcome::Error(xcm::v3::Error::Barrier),
+					..
+				}) => {},
+			]
+		);
+	});
+
+	// No event on the AssetHub message is meant for relay chain
+	AssetHubPolkadot::execute_with(|| {
+		assert_eq!(AssetHubSystem::events().len(), 0);
+	});
+}
+
+#[test]
+fn test_sudo_call_from_asset_hub_to_spiritnet() {
+	MockNetworkPolkadot::reset();
+
+	// create the sovereign account of AssetHub
+	let asset_hub_sovereign_account =
+		Spiritnet::sovereign_account_id_of(Spiritnet::sibling_location_of(AssetHubPolkadot::para_id()));
+
+	let code = vec![];
+
+	let call: DoubleEncoded<()> = <Spiritnet as Parachain>::RuntimeCall::System(frame_system::Call::set_code { code })
+		.encode()
+		.into();
+	let sudo_origin = <AssetHubPolkadot as Parachain>::RuntimeOrigin::root();
+	let parachain_destination: VersionedMultiLocation =
+		ParentThen(Junctions::X1(Junction::Parachain(spiritnet::PARA_ID))).into();
+
+	let weight_limit = WeightLimit::Unlimited;
+	let require_weight_at_most = Weight::from_parts(1600000000000, 200000);
+	let origin_kind = OriginKind::Superuser;
```

Comment:
Also here, all other `OriginKind`s as well.

Reply:
This would be a bit pointless, no? If I select another `OriginKind`, it's obvious that the origin on the target chain isn't transformed to root, which is what I intended to test here.

Reply:
It is obvious to who? It's internal logic of the destination parachain to transform a (location, origin_kind) tuple into a local origin, and there's technically nothing stopping us from (mistakenly) transform anything unwanted to the root origin. It would be a very bad mistake, but it can happen. Hence I think a test covering that would go a long way.

Reply:
In pratice we want to tests that under no circumstances a call to a pallet that requires a `sudo` origin can be dispatched over XCM.

Reply:
In pratice we want to tests that under no circumstances a call to a pallet that requires a `sudo` origin can be dispatched over XCM.

Reply:
In pratice we want to test that under no circumstances a call to a pallet that requires a `sudo` origin can be dispatched over XCM.

Reply:
I see okay I will add the other test cases.

Reply:
I added tests cases for all possible origin kinds: [5f86edf](https://github.com/KILTprotocol/kilt-node/pull/603/commits/5f86edfa7932e4cd58174f351447b35c2c61c29a)

Reply:
I added here all pssible origin kinds: [5f86edf](https://github.com/KILTprotocol/kilt-node/pull/603/commits/5f86edfa7932e4cd58174f351447b35c2c61c29a)